src/commands/getDefaultWebAppToDeploy.ts

```diff
@@ -11,8 +11,9 @@ import { IDeployWizardContext } from "./deploy/IDeployWizardContext";
 
 export async function getDefaultWebAppToDeploy(context: IDeployWizardContext): Promise<WebAppTreeItem | undefined> {
     const defaultWebAppId: string | undefined = getWorkspaceSetting(configurationSettings.defaultWebAppToDeploy, context.workspace.uri.fsPath);
-
-    if (defaultWebAppId && defaultWebAppId !== none) {
+    if (defaultWebAppId === none) {
+        context.telemetry.properties.deployedWithConfigs = 'none';
+    } else if (defaultWebAppId) {
         const defaultWebApp: WebAppTreeItem | undefined = await ext.tree.findTreeItem(defaultWebAppId, context); // resolves to undefined if app can't be found
         if (defaultWebApp) {
             context.deployedWithConfigs = true;
```

Comment:
The name `deployedWithConfigs` doesn't make sense to me. Which configs is it talking about? I would guess defaultWebAppToDeploy _and_ deploySubpath based on the name, but as far as I can tell only `defaultWebAppToDeploy` matters for this. As a possible alternative, we could refactor to have a `webAppSource` property that allows the values `setting`, `rightClick` or `nodePicker`

Reply:
Yeah, agreed. I think back when I made that property, we never talked about using `deploySubpath` settings.  I think I'd prefer a `webAppSource` property as you describe as the subDeployPath stuff should be handled by the `appservice` package now.